src/meta/src/hummock/compaction/picker/trivial_move_compaction_picker.rs

```diff
@@ -37,45 +39,85 @@ impl TrivialMovePicker {
         target_level: usize,
         overlap_strategy: Arc<dyn OverlapStrategy>,
         sst_allowed_trivial_move_min_size: u64,
+        sst_allowed_trivial_move_max_count: usize,
     ) -> Self {
         Self {
             level,
             target_level,
             overlap_strategy,
             sst_allowed_trivial_move_min_size,
+            sst_allowed_trivial_move_max_count,
         }
     }
 
-    pub fn pick_trivial_move_sst(
+    pub fn pick_multi_trivial_move_ssts(
         &self,
         select_tables: &[SstableInfo],
         target_tables: &[SstableInfo],
         level_handlers: &[LevelHandler],
         stats: &mut LocalPickerStatistic,
-    ) -> Option<SstableInfo> {
+    ) -> Option<Vec<SstableInfo>> {
         let mut skip_by_pending = false;
+
+        // means we have already found some sst but not match `sst_allowed_trivial_move_max_count`.
+        let mut skip_by_size = false;
+        let mut result = vec![];
+
+        let mut overlap_info = self.overlap_strategy.create_overlap_info();
         for sst in select_tables {
+            if result.len() >= self.sst_allowed_trivial_move_max_count {
+                break;
+            }
+
+            // find the first sst that can be trivial moved.
             if sst.sst_size < self.sst_allowed_trivial_move_min_size {
+                skip_by_size = true;
+
+                // Stop probing if we have already found some sst to move. And should avoid small sst move to target level.
+                if !result.is_empty() {
+                    break;
+                }
+
                 continue;
             }
 
             if level_handlers[self.level].is_pending_compact(&sst.sst_id) {
                 skip_by_pending = true;
+
+                if !result.is_empty() {
+                    break;
+                }
+
                 continue;
             }
-            let mut overlap_info = self.overlap_strategy.create_overlap_info();
             overlap_info.update(sst);
             let overlap_files_range = overlap_info.check_multiple_overlap(target_tables);
 
             if overlap_files_range.is_empty() {
-                return Some(sst.clone());
+                result.push(sst.clone());
+            } else {
+                // stop probing if we have already found some sst to move.
+                if !result.is_empty() {
+                    break;
+                }
+
+                // reset overlap_info
+                overlap_info.clear();
```

Comment:
minor: call `create_overlap_info` is equivalent to the new method `clear`.

Reply:
`create_overlap_info` has an additional box memory allocation overhead, but this is more intuitive. The semantics of `clear` are a bit unclean, but both work for me.